crates/server/src/api/routes/quest_instances/get.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{api::middlewares::RequiredAuthUser, domain::quests::QuestError};
+use actix_web::{get, web, HttpResponse};
+use quests_db::{
+    core::definitions::{QuestInstance, QuestsDatabase},
+    Database,
+};
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Deserialize, Serialize, ToSchema)]
+pub struct GetQuestInstanceResponse {
+    pub instance: QuestInstance,
+}
+
+/// Get a specific quest instance. Only the Quest Creator is allowed to see the Quest Instances
+#[utoipa::path(
+    params(
+        ("quest_instance_id" = String, description = "Quest Instance UUID")
+    ),
+    responses(
+        (status = 200, description = "Quest Instance", body = GetQuestInstanceResponse),
+        (status = 401, description = "Unathorized"),
+        (status = 403, description = "Forbidden"),
+        (status = 404, description = "Not Found"),
+        (status = 500, description = "Internal Server Error")
+    )
+)]
+#[get("/instances/{quest_instance_id}")]
+pub async fn get_quest_instance(
+    db: web::Data<Database>,
+    quest_instance_id: web::Path<String>,
+    auth_user: RequiredAuthUser,
+) -> HttpResponse {
+    let db = db.into_inner();
+    let quest_instance_id = quest_instance_id.into_inner();
+
+    let RequiredAuthUser { address } = auth_user;
+
+    match db.get_quest_instance(&quest_instance_id).await {
+        Ok(instance) => match db.is_quest_creator(&instance.quest_id, &address).await {
+            Ok(is_creator) if !is_creator => HttpResponse::from_error(QuestError::NotQuestCreator),
+            Ok(_) => HttpResponse::Ok().json(GetQuestInstanceResponse { instance }),
+            Err(err) => {
+                log::error!(
+                    "error on checking if {address} is quest creator of {}",
+                    instance.quest_id
+                );
```

Comment:
Should we log this error? seems a business validation and may can cause flood on logs in we get attacked

Reply:
Should we return Unauthorized on this case?

Reply:
Or maybe return `HttpResponse::from_error(QuestError::NotQuestCreator)`

Reply:
`HttpResponse::from_error(QuestError::NotQuestCreator)` is already considered in the first `Ok()` branch. The function returns an error when the DB fails, not when there is a mismatch.